### PR TITLE
Frontend with type script

### PR DESCRIPTION
**Pull request by Gedzis**

### CODE REVIEW

File: frontend/src/utils/dates.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export function getFormattedDateTime(date = new Date()) {
+  return `${date.getMonth() + 1}/${date.getDate()} ${date.getHours()}:${padLeadingZero(date.getMinutes())}:${padLeadingZero(date.getSeconds())}`;
+}
```
Comment (DonatasD) [inline frontend/src/utils/dates.js]: Could use moment here and make this readable.
```
export function getFormattedDateTime(date = new Date(), format = 'h:mm:ss') {
  return moment(date).format(format);
}
```

File: frontend/src/utils/math.spec.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  });
+
+  describe('addArray', () => {
+    it('returns 0 when passed empty array', () => {
+      expect(addArray([])).toEqual(0);
+    });
+
+    // it('returns null when passed null', () => {
+    //  should.not.exist(MathHelper.addArray(null));
+    // });
```
Comment (justinask7) [inline frontend/src/utils/math.spec.js]: delete?

File: backend/apps/users/serializers.py [hunk 3]
```diff
@@ -1,6 +1,22 @@
+from django.contrib.auth import authenticate
+
 from rest_framework import serializers
 
 
 class LoginSerializer(serializers.Serializer):
     email = serializers.EmailField(max_length=200)
     password = serializers.CharField(max_length=255, required=True, style={'input_type': 'password'})
```
Comment (rmtrimonyte) [inline backend/apps/users/serializers.py]: required=True is default value, you could skip it

File: backend/apps/users/serializers.py [hunk 4]
```diff
@@ -1,6 +1,22 @@
...
+
+    def validate(self, attrs):
+        email = attrs.get('email')
+        password = attrs.get('password')
+
+        user = authenticate(email=email, password=password)
```
Comment (rmtrimonyte) [inline backend/apps/users/serializers.py]: validate function should only do validation on parameters, authenticate() should be perform in create() or in view ? This make serializer not reusable, if in another case validation on parameters and not user authentication is needed

File: frontend/.gitignore [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+# IDEA/Webstorm project files
+.idea
+*.iml
+
+#VSCode metadata
+.vscode
+
+# Mac files
+.DS_Store
```
Comment (justinask7) [inline frontend/.gitignore]: IDE/VSCode ignores maybe should appear at the root .gitignore?

**Pull request merged**